From 9f7f93da768003b7f893281cdda2a8df31b23503 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Fri, 24 Jul 2020 14:37:56 +0200
Subject: [PATCH] waylandsink: fix meson build

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 ext/wayland/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/wayland/meson.build b/ext/wayland/meson.build
index 3c377ee..c79ba7b 100644
--- a/ext/wayland/meson.build
+++ b/ext/wayland/meson.build
@@ -3,6 +3,7 @@ wl_sources = [
     'wlshmallocator.c',
     'wlbuffer.c',
     'wldisplay.c',
+    'wlutils.c',
     'wlwindow.c',
     'wlvideoformat.c',
     'wllinuxdmabuf.c'
-- 
2.27.0

