From e5a8f1203e047c165cfa61053b58a7c06b60f433 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Mon, 28 Jun 2021 20:37:54 +0200
Subject: [PATCH] imx/run.sh: remove bashism

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 imx/run.sh | 51 ++++++++++++++++++++++++---------------------------
 1 file changed, 24 insertions(+), 27 deletions(-)

diff --git a/imx/run.sh b/imx/run.sh
index ebeb7a6e7..d6a7d7a3e 100755
--- a/imx/run.sh
+++ b/imx/run.sh
@@ -5,7 +5,7 @@
 # (c) Basler 2020
 #
 
-RUN_SCRIPT=`realpath -s $0`
+RUN_SCRIPT=`realpath $0`
 RUN_SCRIPT_PATH=`dirname $RUN_SCRIPT`
 echo "RUN_SCRIPT=$RUN_SCRIPT"
 echo "RUN_SCRIPT_PATH=$RUN_SCRIPT_PATH"
@@ -15,15 +15,15 @@ LOCAL_RUN="0" # search modules in /lib/modules and libraries in /usr/lib
 JSON_FILE=""
 LOCAL_RUN="0"
 # an array with the modules to load, insertion order
-declare -a MODULES=("imx8-media-dev" "vvcam-video" "vvcam-dwe" "vvcam-isp")
+MODULES="imx8-media-dev vvcam-video vvcam-dwe vvcam-isp"
 
 USAGE="Usage:\n"
-USAGE+="run.sh -c <isp_config> &\n"
-USAGE+="Supported configurations:\n"
-USAGE+="\tbasler_1080p60    - single basler camera on MIPI-CSI1, 1920x1080, 60 fps\n"
-USAGE+="\tbasler_4k         - single basler camera on MIPI-CSI1, 3840x2160, 30 fps\n"
-USAGE+="\tbasler_1080p60hdr - single basler camera on MIPI-CSI1, 1920x1080, 60 fps, HDR configuration\n"
-USAGE+="\tbasler_4khdr      - single basler camera on MIPI-CSI1, 3840x2160, 30 fps, HDR configuration\n"
+USAGE=$USAGE"run.sh -c <isp_config> &\n"
+USAGE=$USAGE"Supported configurations:\n"
+USAGE=$USAGE"\tbasler_1080p60    - single basler camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+USAGE=$USAGE"\tbasler_4k         - single basler camera on MIPI-CSI1, 3840x2160, 30 fps\n"
+USAGE=$USAGE"\tbasler_1080p60hdr - single basler camera on MIPI-CSI1, 1920x1080, 60 fps, HDR configuration\n"
+USAGE=$USAGE"\tbasler_4khdr      - single basler camera on MIPI-CSI1, 3840x2160, 30 fps, HDR configuration\n"
 
 
 # parse command line arguments
@@ -167,21 +167,18 @@ load_module () {
 
 load_modules () {
 	# remove any previous instances of the modules
-	n=${#MODULES_TO_REMOVE[*]}
-	for (( i = n-1; i >= 0; i-- ))
+	for i in "${MODULES_TO_REMOVE}"
 	do
-		echo "Removing ${MODULES_TO_REMOVE[i]}..."
-		rmmod ${MODULES_TO_REMOVE[i]} &>/dev/null
-		#LSMOD_STATUS=`lsmod | grep "${MODULES[i]}"`
-		#echo "LSMOD_STATUS=$LSMOD_STATUS"
-		if lsmod | grep "${MODULES_TO_REMOVE[i]}" ; then
-			echo "Removing ${MODULES_TO_REMOVE[i]} failed!"
+		echo "Removing $i..."
+		rmmod $i &>/dev/null
+		if lsmod | grep "$i" ; then
+			echo "Removing $i failed!"
 			exit 1
 		fi
 	done
 
 	# and now clean load the modules
-	for i in "${MODULES[@]}"
+	for i in "${MODULES}"
 	do
 		echo "Loading module $i ..."
 		load_module $i
@@ -191,10 +188,10 @@ load_modules () {
 write_default_mode_files
 
 echo "Trying configuration \"$ISP_CONFIG\"..."
-MODULES_TO_REMOVE=( "basler-camera-driver-vvcam" "os08a20" "ov2775" "${MODULES[@]}")
+MODULES_TO_REMOVE="basler-camera-driver-vvcam os08a20 ov2775 ${MODULES}"
 case "$ISP_CONFIG" in
 		basler_4k )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			JSON_FILE="case/config/daA3840_30mc_4K.json"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="DAA3840_30MC_4K.drv"
@@ -205,7 +202,7 @@ case "$ISP_CONFIG" in
 			VVCAM_PARAMS="fourcc=$FOURCC width=3840 height=2160"
 			;;
 		basler_1080p60 )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			JSON_FILE="case/config/daA3840_30mc_1080P.json"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="DAA3840_30MC_1080P.drv"
@@ -216,7 +213,7 @@ case "$ISP_CONFIG" in
 			VVCAM_PARAMS="fourcc=$FOURCC width=1920 height=1080"
 			;;
 		basler_4khdr )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			JSON_FILE="case/config/daA3840_30mc_4K.json"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="DAA3840_30MC_4K.drv"
@@ -227,7 +224,7 @@ case "$ISP_CONFIG" in
 			VVCAM_PARAMS="fourcc=$FOURCC width=3840 height=2160"
 			;;
 		basler_1080p60hdr )
-			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
+			MODULES="basler-camera-driver-vvcam ${MODULES}"
 			JSON_FILE="case/config/daA3840_30mc_1080P.json"
 			CAM_NAME="basler-vvcam"
 			DRV_FILE="DAA3840_30MC_1080P.drv"
@@ -238,7 +235,7 @@ case "$ISP_CONFIG" in
 			VVCAM_PARAMS="fourcc=$FOURCC width=1920 height=1080"
 			;;
 		ov2775_1080p30 )
-			MODULES=("ov2775" "${MODULES[@]}")
+			MODULES="ov2775 ${MODULES}"
 			JSON_FILE="case/config/2006_sensor_dwe.json"
 			CAM_NAME="ov2775"
 			DRV_FILE="ov2775.drv"
@@ -248,7 +245,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		ov2775_1080p30hdr )
-			MODULES=("ov2775" "${MODULES[@]}")
+			MODULES="ov2775 ${MODULES}"
 			JSON_FILE="case/config/2006_sensor_dwe.json"
 			CAM_NAME="ov2775"
 			DRV_FILE="ov2775.drv"
@@ -258,7 +255,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		ov2775_1080p30hdr_native )
-			MODULES=("ov2775" "${MODULES[@]}")
+			MODULES="ov2775 ${MODULES}"
 			JSON_FILE="case/config/2006_sensor_dwe.json"
 			CAM_NAME="ov2775"
 			DRV_FILE="ov2775.drv"
@@ -268,7 +265,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		ov2775_720p )
-			MODULES=("ov2775" "${MODULES[@]}")
+			MODULES="ov2775 ${MODULES}"
 			JSON_FILE="case/config/2006_sensor_dwe.json"
 			CAM_NAME="ov2775"
 			DRV_FILE="ov2775.drv"
@@ -278,7 +275,7 @@ case "$ISP_CONFIG" in
 			write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
 			;;
 		os08a20_4k )
-			MODULES=("os08a20" "${MODULES[@]}")
+			MODULES="os08a20 ${MODULES}"
 			JSON_FILE="case/config/2005_mc_sensor.json"
 			CAM_NAME="os08a20"
 			DRV_FILE="os08a20.drv"
-- 
2.30.2

